.gitlab/stream-ci.yaml

```diff
@@ -0,0 +1,199 @@
+stages:
+  - build   # Test if headers build and install (CMake)
+  - test    # Test if unit tests pass (CTest)
+  - consume # Test if downstream can depend on installs (CMake)
+
+variables:
+  DEPS_DIR:      "$CI_PROJECT_DIR/dependencies"
+  BUILD_DIR:     "$CI_PROJECT_DIR/build"
+  INSTALL_DIR:   "$CI_PROJECT_DIR/install"
+  CMAKE_MINIMUM: "3.0.2"
+  CMAKE_LATEST:  "3.21.2"
+
+########################
+##                    ##
+## Setup environments ##
+##                    ##
+########################
+
+.toolchain-matrix-minimum:
+  image: streamhpc/opencl-sdk-base:ubuntu-18.04-20211006
+  parallel:
+    matrix:
+  # NOTE: Old CMake versions will likely not use cutting-edge compilers
+      - COMPILER: [gcc-7, gcc-8, clang-8, clang-9, clang-10]
+  before_script:
+    - export PATH=/opt/Kitware/CMake/$CMAKE_MINIMUM/bin:$PATH
+
+.toolchain-matrix-latest:
+  image: streamhpc/opencl-sdk-base:ubuntu-18.04-20211006
+  parallel:
+    matrix:
+  # NOTE: Cutting edge CMake versions will likely not use old compilers
+      - COMPILER: [gcc-9, gcc-10, clang-11, clang-12, clang-13]
+  before_script:
+    - export PATH=/opt/Kitware/CMake/$CMAKE_LATEST/bin:$PATH
+
+
+#################
+##             ##
+## Build stage ##
+##             ##
+#################
+
+build:cmake-minimum:
+  extends: .toolchain-matrix-minimum
+  stage: build
+  # NOTE 1: CMake creates build and install folders as needed.
+  # NOTE 2: Lack of space in -H<folder> -B<folder> is important!
+  #         https://stackoverflow.com/a/20611964/1476661
+  script:
+    - cmake
+      -G Ninja
+      -D CMAKE_C_COMPILER=$COMPILER
+      -D CMAKE_BUILD_TYPE=Release
+      -D BUILD_TESTING=ON
+      -D CMAKE_INSTALL_PREFIX=$INSTALL_DIR
+      -B$BUILD_DIR
+      -H$CI_PROJECT_DIR
+    - cmake
+      --build $BUILD_DIR
+      --target install
+  artifacts:
+    paths:
+      - build/
+      - install/
+    exclude:
+      - build/**/CMakeFiles/
+    expire_in: 2 weeks
+
+build:cmake-latest:
+  extends: .toolchain-matrix-latest
+  stage: build
+  script:
+    - cmake
+      -G "Ninja Multi-Config"
+      -D CMAKE_C_COMPILER=$COMPILER
+      -D BUILD_TESTING=ON
+      -D CMAKE_INSTALL_PREFIX=$INSTALL_DIR
+      -B $BUILD_DIR
+      -S $CI_PROJECT_DIR
+    - cmake
+      --build $BUILD_DIR
+      --config Debug
+      --target install
+    - cmake
+      --build $BUILD_DIR
+      --config Release
+      --target install
+  artifacts:
+    paths:
+      - build/
+      - install/
+    exclude:
+      - build/**/CMakeFiles/
+    expire_in: 2 weeks
+
+
+################
+##            ##
+## Test stage ##
+##            ##
+################
+
+## ROCm
```

Comment:
Is this a typo or a copy-paste issue?  I don't see anything here having to do with ROCm.

Reply:
Indeed, copy-pasta error from our own scripts. Removed to avoid confusion.